src/tfvc/commands/findworkspace.ts

```diff
@@ -0,0 +1,88 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+"use strict";
+
+import { IExecutionResult, ITfvcCommand, IWorkspace } from "../interfaces";
+import { ArgumentBuilder } from "./argumentbuilder";
+
+/**
+ * This command only returns a partial workspace object that allows you to get the name and server.
+ * To get the entire workspace object you should call GetWorkspace with the workspace name.
+ * (This is one of the only commands that expects to be a strictly local operation - no server calls - and so does not
+ * take a server context object in the constructor)
+ */
+export class FindWorkspace implements ITfvcCommand<IWorkspace> {
+    private _localPath: string;
+
+    constructor(localPath: string) {
+        this._localPath = localPath;
+    }
+
+    getArguments(): string[] {
+        return new ArgumentBuilder("workfold")
+            .build();
+    }
+
+    getOptions(): any {
+        return { cwd: this._localPath };
+    }
+
+    /**
+     * Parses the output of the workfold command. (NOT XML)
+     * SAMPLE
+     * Access denied connecting to TFS server https://account.visualstudio.com/ (authenticating as Personal Access Token)  <-- line is optional
+     * =====================================================================================================================================================
+     * Workspace:  MyNewWorkspace2
+     * Collection: http://java-tfs2015:8081/tfs/
+     * $/tfsTest_01: D:\tmp\test
+     */
+    parseOutput(executionResult: IExecutionResult): IWorkspace {
+        const stdout = executionResult.stdout;
+
+        // Find the workspace name and collectionUrl
+        const lines = stdout.replace("\r\n", "\n").split("\n");
+        let workspaceName: string = "";
+        let collectionUrl: string = "";
+        let equalsLineFound: boolean = false;
+
+        for (let i = 0; i <= lines.length; i++) {
+            const line = lines[i];
+            if (!line) {
+                continue;
+            }
+
+            if (line.startsWith("==========")) {
+                equalsLineFound = true;
+                continue;
+            } else if (!equalsLineFound) {
+                continue;
+            }
+
+            if (line.startsWith("Workspace:")) {
+                workspaceName = this.getValue(line);
+            } else if (line.startsWith("Collection:")) {
+                collectionUrl = this.getValue(line);
+            }
+        }
+
+        const workspace: IWorkspace = {
+            name: workspaceName,
+            server: collectionUrl
+        };
+
+        return workspace;
+    }
+
+    private getValue(line: string): string {
```

Comment:
A comment would be great as to what the incoming line looks like (presumably there's a ':' somewhere :-).